src/Themes/ui/ColorPicker/index.tsx

```diff
@@ -0,0 +1,240 @@
+import { ColorPreview } from "./ColorPreview";
+import { HexInput, formatRGBtoHEX } from "./HexInput";
+import { HueSlider } from "./HueSlider";
+import { RGBDigit } from "./RGBDigit";
+import { SVCanvas } from "./SVCanvas";
+import React, { Dispatch, useMemo, useReducer, useState } from "react";
+import { useTheme } from "@mui/material/styles";
+import { Modal } from "../../../ui/React/Modal";
+import Button from "@mui/material/Button";
+
+export type HSV = {
+  hue: number;
+  sat: number;
+  val: number;
+};
+
+export type RGB = {
+  r: number;
+  g: number;
+  b: number;
+  a?: number;
+};
+
+function normalize(color: HSV | RGB) {
+  const isRGB = (c: RGB | HSV): c is RGB => Object.hasOwn(color, "r");
+  const rgb: RGB = isRGB(color) ? color : HSVtoRGB(color);
+  const hsv: HSV = !isRGB(color) ? color : RGBtoHSV(color);
+  return { rgb, hsv };
+}
+
+type OpenColorPickerButtonProps = {
+  color: string;
+  title: string;
+  onColorChange: (color: string) => void;
+};
+
+function parseHEXtoRGB(hex: string): RGB {
+  console.log(hex);
+
+  if (hex.length == 4) {
+    return {
+      r: Number.parseInt(`${hex[1]}${hex[1]}`, 16),
+      g: Number.parseInt(`${hex[2]}${hex[2]}`, 16),
+      b: Number.parseInt(`${hex[3]}${hex[3]}`, 16),
+    };
+  }
+
+  if (hex.length == 7 || hex.length == 9) {
+    return {
+      r: Number.parseInt(`${hex[1]}${hex[2]}`, 16),
+      g: Number.parseInt(`${hex[3]}${hex[4]}`, 16),
+      b: Number.parseInt(`${hex[5]}${hex[6]}`, 16),
+      a: Number.parseInt(`${hex[7]}${hex[8]}`, 16),
+    };
+  }
+
+  throw new Error("Invalid hex string");
```

Comment:
What happens when this Error is thrown? Should we perhaps be more lenient and return a default color?

Reply:
This error shouldn't be reachable in production code. It only receives input from outside the component (current selected color) or already validated input from the HexInput. Both of those should have mechanisms already in place in case of invalid formatting so this is mainly to keep TS happy. 
I'm against returning a default color because that makes it unclear when an error actually happens. If there is a case that throws this error that needs to be fixed, not hidden